src/casanovoutils/filter_spectra.py

```
@@ -0,0 +1,225 @@
+        n_total += 1
+
+        # --- 1. Missing or empty SEQ ----------------------------------------
+        seq = spectrum["params"].get("seq", "")
+        if not seq:
+            n_no_seq += 1
+            continue
+
+        # --- 2. Unknown tokens in sequence ----------------------------------
+        try:
+            tokens = _seq_to_tokens(seq)
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)